traffic_monitor/manager/manager.go

```diff
@@ -45,6 +46,8 @@ import (
 func Start(opsConfigFile string, cfg config.Config, appData config.StaticAppData, trafficMonitorConfigFileName string) error {
 	toSession := towrap.ITrafficOpsSession(towrap.NewTrafficOpsSessionThreadsafe(nil, cfg.CRConfigHistoryCount, cfg))
 
+	url6Regex := regexp.MustCompile(`/\d+`)
```

Comment:
Regexes are typically much slower than simple string manipulation, and TM is highly performance-sensitive. This isn't actually in the request path; but it does get called for every server, every time a new config is received.

Would you object to changing the usage to a string function? E.g.
 
```
// ipv6CIDRStrToAddr takes an IPv6 CIDR string, e.g. `2001:DB8::1/32` returns `2001:DB8::1`.
// It does not verify cidr is a valid CIDR or IPv6. It only removes the first slash and everything after it, for performance.
func ipv6CIDRStrToAddr(cidr string) string {
	i := strings.Index(cidr, `/`)
	if i == -1 {
		return cidr
	}
	return cidr[:i]
}
```
https://play.golang.org/p/1tLFJ7P1C91

Reply:
If you're mostly just interested in getting the IP address part of a CIDR-notation network then you could also just get the first return value of [`net.ParseCIDR`](https://godoc.org/net#ParseCIDR)

Reply:
`net.ParseCIDR` is expensive too, though, and doing a lot of work that we don't need here.

Reply:
I wanted to be sure, so I wrote a benchmark:

```
package main

import (
	"math/rand"
	"net"
	"regexp"
	"strconv"
	"strings"
	"testing"
)

func BenchmarkRemoveCIDRFunc(b *testing.B) {
	cidrStrs := randCIDRStrs(b.N)
	ipStrs := make([]string, len(cidrStrs), len(cidrStrs))

	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		ipStrs[i] = ipv6CIDRStrToAddr(cidrStrs[i])
	}
}

func BenchmarkRemoveCIDRRegex(b *testing.B) {
	re := regexp.MustCompile(`/\d+`)
	cidrStrs := randCIDRStrs(b.N)
	ipStrs := make([]string, len(cidrStrs), len(cidrStrs))

	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		ipStrs[i] = re.ReplaceAllString(cidrStrs[i], "")
	}
}

func BenchmarkRemoveCIDRIP(b *testing.B) {
	cidrStrs := randCIDRStrs(b.N)
	ipStrs := make([]string, len(cidrStrs), len(cidrStrs))

	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		ip, _, err := net.ParseCIDR(cidrStrs[i])
		if err != nil {
			b.Fatal("not an IP: '" + cidrStrs[i] + "'")
		}
		ipStrs[i] = ip.String()
	}
}

func randCIDRStrs(n int) []string {
	ips := []net.IP{}
	for i := 0; i < n; i++ {
		ip := make([]byte, 16, 16)
		rand.Read(ip) // math/rand.Read is documented to always return a nil error.
		ips = append(ips, net.IP(ip))
	}

	ipStrs := []string{}
	for _, ip := range ips {
		ipStrs = append(ipStrs, ip.String()+`/`+strconv.Itoa(rand.Intn(129)))
	}

	return ipStrs
}

// ipv6CIDRStrToAddr takes an IPv6 CIDR string, e.g. `2001:DB8::1/32` returns `2001:DB8::1`.
// It does not verify cidr is a valid CIDR or IPv6. It only removes the first slash and everything after it, for performance.
func ipv6CIDRStrToAddr(cidr string) string {
	i := strings.Index(cidr, `/`)
	if i == -1 {
		return cidr
	}
	return cidr[:i]
}
```

```
$ go test -bench=.
goos: darwin
goarch: amd64
pkg: github.com/apache/trafficcontrol/traffic_monitor
BenchmarkRemoveCIDRFunc-16      63100605                23.7 ns/op
BenchmarkRemoveCIDRRegex-16      4373604               290 ns/op
BenchmarkRemoveCIDRIP-16         2069672               528 ns/op
PASS
ok      github.com/apache/trafficcontrol/traffic_monitor        37.686s
```

Reply:
Sure but if the result of stripping is used for TCP dials then you're paying that cost anyway. Idk how it's used later on, just wanted to throw that out there.

Reply:
It isn't. But that's not a bad idea, might be worth looking into. I don't know how the cost of parsing the IP compares to the request cost; but we do enough requests, and use enough CPU, it might be significant.

Reply:
done